### PR TITLE
Validate readiness of ApplicationDraft

### PR DESCRIPTION
**Pull request by ramonh**
This pull request enables the validation of `ApplicationDraft`'s `ready` state.

This brings us a step closer to completing #163.

### CODE REVIEW

File: spec/controllers/application_drafts_controller_spec.rb [hunk 1]
```diff
@@ -168,7 +168,7 @@
 
       context 'for a valid draft' do
         before do
-          allow_any_instance_of(ApplicationDraft).to receive(:valid?).with(:apply).and_return(true)
+          allow_any_instance_of(ApplicationDraft).to receive(:ready?).and_return(true)
```
Comment (carpodaster) [inline spec/controllers/application_drafts_controller_spec.rb]: The tests should pass with the old stub in place since `#ready?` only forwards to that.

Comment (carpodaster) [inline spec/controllers/application_drafts_controller_spec.rb]: In fact, it _could_ fix the currently failing spec :wink:

File: app/models/application_draft.rb [hunk 2]
```diff
@@ -66,7 +66,7 @@ def role_for(user)
   end
 
   def ready?
-    false # valid?(:apply)
+    valid?(:apply)
```
Comment (carpodaster) [inline app/models/application_draft.rb]: We can't merge this PR yet as that would enable the submit button in app/views/application_drafts/new.html.slim:63 but we do not have the `ApplicationDrafts#apply` controller action yet.

File: spec/controllers/application_drafts_controller_spec.rb (hunk 1 above)
Comment (ramonh) [inline spec/controllers/application_drafts_controller_spec.rb]: This is true! However, the spec indeed fails as we're missing the ability to create an `Application` from an `ApplicationDraft`. I'll investigate.

File: app/models/application_draft.rb (hunk 2 above)
Comment (ramonh) [inline app/models/application_draft.rb]: Ah yes, makes sense. I'll look into adding this.

**Comment by ramonh** [comment]
I think it makes sense to close this PR for now, and focus on getting the `ApplicationDrafts#apply` controller action working.

**Pull request closed without merging**